src/databricks/sql/utils.py

```diff
@@ -470,3 +474,66 @@ def _create_arrow_array(t_col_value_wrapper, arrow_type):
             result[i] = None
 
     return pyarrow.array(result, type=arrow_type)
+
+class DbSqlType(Enum):
+    STRING = "STRING"
+    DATE = "DATE"
+    TIMESTAMP = "TIMESTAMP"
+    FLOAT = "FLOAT"
+    DECIMAL = "DECIMAL"
+    INTEGER = "INTEGER"
+    BIGINT = "BIGINT"
+    SMALLINT = "SMALLINT"
+    TINYINT = "TINYINT"
+    BOOLEAN = "BOOLEAN"
+    INTERVAL_MONTH = "INTERVAL MONTH"
+    INTERVAL_DAY = "INTERVAL DAY"
+
+class DbSqlParameter:
+    name: str
+    value: Any
+    type: DbSqlType
+    def __init__(self,name="",value=None,type=None):
+        self.name = name
+        self.value = value
+        self.type = type
+        
+        
+def named_parameters_to_dbsqlparams(parameters: Any):
+    dbsqlparams = []
+    if(type(parameters) is dict):
+        # Support legacy impl
+        for name,parameter in parameters.items():
+            dbsqlparams.append(DbSqlParameter(name=name,value=parameter))
+    else:
+        for parameter in parameters:
+            if(type(parameter) is DbSqlParameter):
+                dbsqlparams.append(parameter)
+            else:
+                dbsqlparams.append(DbSqlParameter(value=parameter))
+    return dbsqlparams
+
+def infer_types(params: list[DbSqlParameter]):
+    for param in params:
+        if(not param.type):
+            if(type(param.value) is str):
+                param.type = DbSqlType.STRING
+            elif(type(param.value) is int):
+                param.type = DbSqlType.INTEGER
+            elif(type(param.value) is float):
+                param.type = DbSqlType.FLOAT
+            elif(type(param.value) is datetime.datetime):
+                param.type = DbSqlType.TIMESTAMP
+            elif(type(param.value) is bool):
+                param.type = DbSqlType.BOOLEAN
+        if(param.value):
```

Comment:
`if` is not a function in python, just a keyword. This code _works_ because the parentheses create a tuple which is still truthy or falsey. But the syntax isn't pythonic.

```suggestion
        if param.value:
```

It's also not clear what the failure modes are for this code. What if `param.value` is `None`? What if `param.value` evaluates falsey such as the integer `0` or the singleton `False`?

Reply:
ya, this is not a useful check, I've changed it to just stringify